emily_cron/app/services/deposit_processor.py

```diff
@@ -203,6 +207,55 @@ def process_expired_locktime(
 
         return updates
 
+    def process_long_pending(
+        self,
+        enriched_deposits: list[EnrichedDepositInfo],
+    ) -> list[DepositUpdate]:
+        """Process long-pending transactions.
+
+        Args:
+            enriched_deposits: List of enriched deposit information
+
+        Returns:
+            list[DepositUpdate]: List of deposit updates
+        """
+        updates = []
+
+        current_time = int(datetime.now().timestamp())
+
+        long_pending_txs = []
+        for tx in enriched_deposits:
+            # Only check pending transactions
+            if tx.status != RequestStatus.PENDING.value:
+                continue
+            # that we can't find via the mempool API (it might have been dropped)
+            if tx.in_mempool:
+                continue
+
+            try:
+                # and have been pending for too long
+                if current_time - tx.deposit_last_update > settings.MAX_UNCONFIRMED_TIME:
+                    long_pending_txs.append(tx)
+            except (RequestException, ValueError, JSONDecodeError) as e:
+                logger.warning(
+                    f"Could not check pending status for deposit {tx.bitcoin_txid} due to "
+                    f"API error fetching block {tx.last_update_block_hash}: {e}. Skipping."
+                )
```

Comment:
Why is this in a `try-except` block?

Reply:
Ahhh, I see. Would it be better to put LRU caching on the `HiroAPI.get_stacks_block` call instead of on an instance of the `DepositInfo` class?